Modules/_testcapimodule.c

```diff
@@ -3746,6 +3746,85 @@ static PyTypeObject MyList_Type = {
     MyList_new,                                 /* tp_new */
 };
 
+typedef struct {
+    PyObject_HEAD
+    PyObject *obj;
+    long references;
+} testBufObject;
+
+static PyObject *
+testbuf_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
+{
+    testBufObject *self = (testBufObject *)type->tp_alloc(type, 0);
+    if (self == NULL) {
+        return NULL;
+    }
+    self->obj = PyBytes_FromString("test");
+    self->references = 0;
+    return (PyObject *)self;
+}
+
+static int
+testbuf_traverse(testBufObject *self, visitproc visit, void *arg)
+{
+    Py_VISIT(self->obj);
+    return 0;
+}
+
+static int
+testbuf_clear(testBufObject *self)
+{
+    Py_CLEAR(self->obj);
+    return 0;
+}
+
+static void
+testbuf_dealloc(testBufObject *self)
+{
+    PyObject_GC_UnTrack(self);
+    Py_XDECREF(self->obj);
+    Py_TYPE(self)->tp_free((PyObject *) self);
+}
+
+static int
+testbuf_getbuf(testBufObject *self, Py_buffer *view, int flags)
+{
+    int buf = PyObject_GetBuffer(self->obj, view, flags);
+    Py_SETREF(view->obj, Py_NewRef((PyObject *)self));
+    self->references++;
+    return buf;
+}
+
+static void
+testbuf_releasebuf(testBufObject *self, Py_buffer *view)
+{
+    self->references--;
+    assert(self->references >= 0);
+}
+
+static PyBufferProcs testbuf_as_buffer = {
+    .bf_getbuffer = (getbufferproc) testbuf_getbuf,
+    .bf_releasebuffer = (releasebufferproc) testbuf_releasebuf,
+};
+
+static struct PyMemberDef testbuf_members[] = {
+    {"references", T_LONG, offsetof(testBufObject, references), READONLY},
+    {NULL},
+};
+
+static PyTypeObject testBufType = {
```

Comment:
I think this can go into https://github.com/python/cpython/tree/main/Modules/_testcapi

Reply:
You mean a new file?

Reply:
yes

Reply:
Done